bundles/binding/org.openhab.binding.fatekplc/src/main/java/org/openhab/binding/fatekplc/internal/FatekPLCSlave.java

```diff
@@ -0,0 +1,157 @@
+import org.openhab.core.items.ItemNotFoundException;
+import org.openhab.core.items.ItemRegistry;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.osgi.service.cm.ConfigurationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.s4u.jfatek.FatekException;
+import com.github.s4u.jfatek.FatekPLC;
+import com.github.s4u.jfatek.FatekReadMixDataCmd;
+import com.github.s4u.jfatek.io.FatekIOException;
+import com.github.s4u.jfatek.registers.Reg;
+import com.github.s4u.jfatek.registers.RegValue;
+
+/**
+ * Implementation of one fatek slave.
+ *
+ * @author Slawomir Jaranowski
+ * @since 1.9.0
+ */
+public class FatekPLCSlave {
+
+	/**
+	 * Main logger.
+	 */
+	private static final Logger logger = LoggerFactory.getLogger(FatekPLCSlave.class);
+
+	// OSGI Service
+	private ItemRegistry itemRegistry;
+
+	private EventPublisher eventPublisher;
+
+	private String name;
+
+	private FatekPLC fatekPLC = null;
+
+	public FatekPLCSlave(String name, EventPublisher eventPublisher, ItemRegistry itemRegistry) {
+		this.name = name;
+		this.eventPublisher = eventPublisher;
+		this.itemRegistry = itemRegistry;
+	}
+
+	public String getName() {
+		return name;
+	}
+
+	public void disconnect() throws FatekIOException {
+
+		if (fatekPLC != null) {
+			fatekPLC.close();
+		}
+	}
+
+	/**
+	 * Configure slave property
+	 *
+	 * @param name property name
+	 * @param value config value
+	 * @throws ConfigurationException if something wrong with configuration, eg. Unknown property
+	 */
+	public void configure(String name, Object value) throws ConfigurationException {
+
+		logger.debug("Configure item: {} to {}", name, value);
+
+		if ("connectionUri".equals(name)) {
+			try {
+				disconnect();
+				fatekPLC = new FatekPLC((String) value);
+				logger.info("New Fatek PLC connectionUri={}", value);
+			} catch (Exception e) {
+				fatekPLC = null;
+				throw new ConfigurationException(name, e.getMessage());
+			}
+		} else {
+			throw new ConfigurationException(name, "Unknown property");
+		}
+
+		logger.debug("Configure item: {} end", name);
+	}
+
+	/**
+	 * Read current data from Fatek PLC and publish new state on OpenHab bus.
+	 *
+	 * @param items items to updates
+	 * @throws FatekIOException
+	 * @throws FatekException
+	 */
+	public void updateItems(List<FatekPLCItem> items) throws FatekIOException, FatekException {
+
+		// collect unique register to read from PLC
+		Set<Reg> regsToUpdate = new HashSet<>();
+		for (FatekPLCItem item : items) {
+			regsToUpdate.addAll(item.getRegs());
+		}
+
+		Map<Reg, RegValue> response;
+		synchronized (fatekPLC) {
+			// read register
+			response = new FatekReadMixDataCmd(fatekPLC, regsToUpdate).send();
+		}
+
+		// distribute read data
+		for (FatekPLCItem fatekItem : items) {
+
+			State newState = fatekItem.getState(response);
+			State currentState;
+			Item item;
+			try {
+				item = itemRegistry.getItem(fatekItem.getItemName());
```

Comment:
Looking at the 1..1 cardinality of the service [here](https://github.com/slawekjaranowski/openhab/blob/fatekplc/bundles/binding/org.openhab.binding.fatekplc/OSGI-INF/binding.xml#L31), it looks to me like itemRegistry cannot be `null` since the dependent service is mandatory (cardinality starts with 1, not 0).  Did I miss something, @ssalonen ?